src/shared/toggle_panel.rs

```diff
@@ -0,0 +1,288 @@
+use makepad_widgets::*;
+
+live_design! {
+    import makepad_widgets::base::*;
+    import makepad_widgets::theme_desktop_dark::*;
+    import makepad_draw::shader::std::*;
+
+    ICON_CLOSE_PANEL = dep("crate://self/resources/icons/close_left_panel.svg")
+    ICON_OPEN_PANEL = dep("crate://self/resources/icons/open_left_panel.svg")
+
+    FadeView = <CachedView> {
+        draw_bg: {
+            instance opacity: 1.0
+
+            fn pixel(self) -> vec4 {
+                let color = sample2d_rt(self.image, self.pos * self.scale + self.shift) + vec4(self.marked, 0.0, 0.0, 0.0);
+                return Pal::premul(vec4(color.xyz, color.w * self.opacity))
+            }
+        }
+    }
+
+    CustomButton = <Button> {
```

Comment:
Just a copy of MoxinButton to match moxin styles when porting this outside, as discussed internally.